Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityOptimizeWindow.cs

```diff
@@ -0,0 +1,607 @@
+        private bool enableDepthBufferSharing = true;
+        private bool enable16BitDepthBuffer = true;
+
+        // Scene Optimizations
+        private bool disableRealtimeGlobalIllumination = true;
+        private bool disableBakedGlobalIllumination = true;
+        private static DateTime? lastAnalyzedTime = null;
+        private Light[] sceneLights;
+        private const uint TopListSize = 5;
+        private long totalActivePolyCount, totalInActivePolyCount = 0;
+        private long totalPolyCount
+        {
+            get
+            {
+                return totalActivePolyCount + totalInActivePolyCount;
+            }
+        }
+        private string TotalPolyCountStr, TotalActivePolyCountStr, TotalInactivePolyCountStr;
+        private GameObject[] LargestMeshes = new GameObject[TopListSize];
+        private int[] LargestMeshSizes = new int[TopListSize];
+
+        // Shader Optimizations
+        private bool showDiscoveredMaterials = true;
+        private bool onlyUnityShader = true;
+        private Shader replacementShader;
+        private Shader unityStandardShader;
+        private Shader errorShader;
+        private List<Material> discoveredMaterials = new List<Material>();
+
+        // Internal structure to easily search mesh polycounts in scene
+        private struct MeshNode
+        {
+            public int polycount;
+            public MeshFilter filter;
+        }
+
+        protected enum PerformanceTarget
+        {
+            AR_Headsets,
+            VR_Standalone,
+            VR_Tethered,
+        };
+
+        // NOTE: These array must match the number of enums in PerformanceTarget
+        private readonly string[] PerformanceTargetDescriptions = {
+            "Suggest performance optimizations for AR devices with mobile class specifications",
+            "Suggest performance optimizations for mobile VR devices with mobile class specifications",
+            "Suggest performance optimizations for VR devices tethered to a PC" };
+
+        private readonly int[] SceneLightCountMax = { 1, 2, 4 };
+
+        [SerializeField]
+        private PerformanceTarget PerfTarget = PerformanceTarget.AR_Headsets;
+
+        [SerializeField]
+        private Texture2D logoLightTheme = null;
+
+        [SerializeField]
+        private Texture2D logoDarkTheme = null;
+
+        protected static GUIStyle HelpIconStyle;
+        protected static GUIStyle BoldLargeTitle;
+
+        [MenuItem("Mixed Reality Toolkit/Utilities/Optimize Window", false, 0)]
+        public static void OpenWindow()
+        {
+            // Dock it next to the Scene View.
+            var window = GetWindow<MixedRealityOptimizeWindow>(typeof(SceneView));
+            window.StartUp();
+            window.Show();
+        }
+
+        public void StartUp()
+        {
+            FindShaders();
+
+            string assetPath = "StandardAssets/Textures";
+
+            if (logoLightTheme == null)
```

Comment:
Tangent: we really need to move this sort of thing into an MRTK editor GUI utility.

Reply:
I know :) I was going to do after all my PRs had merged and just do one pass of making things cleaner/more re-usable across components. But I can add the utility now and then just do a follow PR on other stuff to re-use that